toy_problem/src/main.cpp

```diff
@@ -83,6 +83,9 @@ RobotState estimate(const RobotState& initial_state) const final
   state.p = graph.get_waypoint(_dropoff_waypoint).get_location();
   const auto start_time = std::chrono::steady_clock::now() +
     rmf_traffic::time::from_seconds(initial_state.finish_time);
+
+  //(TODO): Duplicate this? Add another planner to plan from location to
+  //  _pickup_waypoint ? 
```

Comment:
Should we add a second planner here? 
`initial location`--------plan1----------  `_pickupwaypoint` ----------plan2----------> `dropoff_waypoint`

Having ` std::move(location)};` as arg in `Planner::Start` assumes the lane is unconstrained

Reply:
Yeah you're right. This was oversight on my behalf.
We should determine the index of the waypoint in the graph nearest to the robot's current location and use that as initial waypoint for plan1. Do you wanna add make the changes in this branch?

Reply:
Quick question, what function should I use to find the nearest graph index with the `location` as input?

Reply:
The full control fleet adapter has an `estimate_waypoint` implementation that you can use as a reference to implement your own here
https://github.com/osrf/rmf_core/blob/6ee8fb3c632a2355256df652c0995918ba45a913/rmf_fleet_adapter/src/full_control/estimation.cpp#L270-L282

Reply:
I've added the function to `master` 
https://github.com/Yadunund/task_planner/blob/15122e28fbdcaa7f939ed64e2cbad2342aa7ac82/toy_problem/src/main.cpp#L40-L58